test/sql/distinct_sql_test.cpp

```diff
@@ -175,5 +175,43 @@ TEST_F(DistinctSQLTests, DistinctStarTest) {
   txn_manager.CommitTransaction(txn);
 }
 
+TEST_F(DistinctSQLTests, DistinctDateTimeTest) {
+  auto &txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  catalog::Catalog::GetInstance()->CreateDatabase(DEFAULT_DB_NAME, txn);
+  txn_manager.CommitTransaction(txn);
+
+  TestingSQLUtil::ExecuteSQLQuery(
+      "CREATE TABLE test(a INT, b TIMESTAMP);");
+
+  // Insert tuples into table
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO test VALUES (1, '2016-06-22 19:10:25-07');");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO test VALUES (1, '2017-06-22 19:10:25-07');");
+  TestingSQLUtil::ExecuteSQLQuery(
+      "INSERT INTO test VALUES (1, '2016-06-22 19:10:25-07');");
+
+  std::vector<StatementResult> result;
+  std::vector<FieldInfo> tuple_descriptor;
+  std::string error_message;
+  int rows_changed;
+
+  TestingSQLUtil::ExecuteSQLQuery("SELECT DISTINCT b FROM test;", result,
+                                  tuple_descriptor, rows_changed,
+                                  error_message);
+
+  // Check the return value
+  EXPECT_EQ(0, rows_changed);
+  EXPECT_EQ(2, result.size() / tuple_descriptor.size());
+  EXPECT_EQ("2016-06-22 19:10:25.000000-07", TestingSQLUtil::GetResultValueAsString(result, 0));
+  EXPECT_EQ("2017-06-22 19:10:25.000000-07", TestingSQLUtil::GetResultValueAsString(result, 1));
```

Comment:
You cannot assume the output will be in this order, but that isn't your fault since all tests in this suite make the same mistake. If you can, try to correct the other tests to check for **set equality** (rather than assuming a deterministic ordering). Take a look at [this](https://github.com/cmu-db/peloton/blob/master/test/sql/optimizer_sql_test.cpp#L114) for an example.